From: Hannes Reinecke <hare@suse.de>
Date: Wed, 13 Nov 2013 12:44:41 +0100
Subject: scsi_dh_alua: correctly terminate target port strings
References: bnc#708296
Patch-Mainline: posted to linux-scsi

The target port identifier is not necessarily NULL terminated,
so we need to do this by hand.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/device_handler/scsi_dh_alua.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/scsi/device_handler/scsi_dh_alua.c b/drivers/scsi/device_handler/scsi_dh_alua.c
index 9739602..308ae1a 100644
--- a/drivers/scsi/device_handler/scsi_dh_alua.c
+++ b/drivers/scsi/device_handler/scsi_dh_alua.c
@@ -517,6 +517,9 @@ static int alua_vpd_inquiry(struct scsi_device *sdev, struct alua_dh_data *h)
 				target_id_size = d[3];
 				target_id = d + 4;
 				strncpy(target_id_str, d + 4, 256);
+				if (target_id_size > 255)
+					target_id_size = 255;
+				target_id_str[target_id_size] = '\0';
 			}
 			break;
 		default:
-- 
1.7.12.4

